### DISCUSSION TITLE
Unknown model (convnext_xxlarge)

### DISCUSSION THREAD

**Discussion by orrzohar** [mlfoundations/open_clip#494] [Q&A]
It seems like convnext_xxlarge is not recognized for some reason: 

 .>tmp = open_clip.list_pretrained()[-5]

.>tmp 
('convnext_xxlarge', 'laion2b_s34b_b82k_augreg_soup')

.>open_clip.create_model_and_transforms(tmp[0], pretrained=tmp[1])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/orrzohar/anaconda3/envs/LOVM/lib/python3.10/site-packages/open_clip/factory.py", line 292, in create_model_and_transforms
    model = create_model(
  File "/home/orrzohar/anaconda3/envs/LOVM/lib/python3.10/site-packages/open_clip/factory.py", line 194, in create_model
    model = CLIP(**model_cfg, cast_dtype=cast_dtype)
  File "/home/orrzohar/anaconda3/envs/LOVM/lib/python3.10/site-packages/open_clip/model.py", line 190, in __init__
    self.visual = _build_vision_tower(embed_dim, vision_cfg, quick_gelu, cast_dtype)
  File "/home/orrzohar/anaconda3/envs/LOVM/lib/python3.10/site-packages/open_clip/model.py", line 90, in _build_vision_tower
    visual = TimmModel(
  File "/home/orrzohar/anaconda3/envs/LOVM/lib/python3.10/site-packages/open_clip/timm_model.py", line 53, in __init__
    self.trunk = timm.create_model(model_name, pretrained=pretrained, **timm_kwargs)
  File "/home/orrzohar/anaconda3/envs/LOVM/lib/python3.10/site-packages/timm/models/factory.py", line 67, in create_model
    raise RuntimeError('Unknown model (%s)' % model_name)
RuntimeError: Unknown model (convnext_xxlarge)

**Comment by rwightman** [comment]
@orrzohar timm still needs to be a 0.8.x pre-release (`pip install --pre timm`) for that model but didn't want to make it a hard dep as not everyone cares about those

**Reply by sina-nabavi** [reply]
I am still facing this issue. I have `timm=1.0.3` but unable to load  `convnext_large` and `convnext_xxlarge`

```
model, _, preprocess = open_clip.create_model_and_transforms('convnext_large_d_320', pretrained='laion2b_s29b_b131k_ft_soup')
tokenizer = open_clip.get_tokenizer('convnext_large')

model, _, preprocess = open_clip.create_model_and_transforms('convnext_xxlarge', pretrained='laion2b_s34b_b82k_augreg_soup')
tokenizer = open_clip.get_tokenizer('convnext_xxlarge')
```
Both return None

**Reply by rwightman** [reply]
@sina-nabavi I'd check versions in your actual environments as it's common for people to be using wrong versions without realizing. Those commands work fine with everything at the latest.
```
print(timm.__version__)
print(open_clip.version.__version__)
```

**Reply by sina-nabavi** [reply]
> @sina-nabavi I'd check versions in your actual environments as it's common for people to be using wrong versions without realizing. Those commands work fine with everything at the latest.
> 
> ```
> print(timm.__version__)
> print(open_clip.version.__version__)
> ```

I get `1.0.3` and `2.24.0` respectively which are the latest versions but still cannot fetch the models `convnext_large` and `convnext_xxlarge`.

**Reply by rwightman** [reply]
@sina-nabavi and this is in the exact same environment, same runtime that you're running thoes open_clip commands? not a different shell, or in an IDE vs not, or in a different notebook, etc?

**Reply by sina-nabavi** [reply]
> @sina-nabavi and this is in the exact same environment, same runtime that you're running thoes open_clip commands? not a different shell, or in an IDE vs not, or in a different notebook, etc?

No. I'm running them from the same shell in the same environment.

**Reply by rwightman** [reply]
I just installed a fresh env, ran the command, no issues. So I've got nothing to go on. You could try calling setup_default_logging() in timm so you see log output, might be a clue there...

```
conda install pytorch torchvision torchaudio pytorch-cuda=12.1 python=3.11 -c pytorch -c nvidia
pip install timm open_clip_torch
...
```

In Python:
```
>>> import open_clip
>>> import timm
>>> timm.utils.setup_default_logging()
>>> model, _, preprocess = open_clip.create_model_and_transforms('convnext_large_d_320', pretrained='laion2b_s29b_b131k_ft_soup')
INFO:root:Loaded convnext_large_d_320 model config.
Loaded convnext_large_d_320 model config.
INFO:root:Loading pretrained convnext_large_d_320 weights (laion2b_s29b_b131k_ft_soup).
Loading pretrained convnext_large_d_320 weights (laion2b_s29b_b131k_ft_soup).
>>> model.logit_scale
Parameter containing:
tensor(4.6052, requires_grad=True)
```
